### PR TITLE
Update to latest BAAS test server.

### PR DESCRIPTION
**Pull request by cmelchior**

### CODE REVIEW

**Comment by cmelchior** [comment]
Ready for review

File: realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/admin/ServerAdmin.kt [hunk 1]
```diff
@@ -100,28 +101,65 @@ class ServerAdmin(private val app: App) {
...
+        val dep = deployments[0] as JSONObject
+        if (dep.getString("status") != "successful") {
+            RealmLog.error("Failed to deploy: ${dep.toString(4)}")
+        }
+
+        // Work-around for /deployments reporting success, but /register still failing.
+        SystemClock.sleep(5000)
+    }
+
+    fun printEmailAuthConfig() {
```
Comment (rorbech) [inline realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/admin/ServerAdmin.kt:145]: Not used? And what is it for? Just local debug?

**Review by rorbech — COMMENTED** [review]
It is a bit unclear whether the workaround for the race is working or just trying to reduce it. I guess it works for me, if it succeeds to a reasonably level. But given previous flaky scenarios we should maybe hesitate to merge. Is it on our side, have somebody looked into the server log and/or is there an issue to track this.

File: realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/admin/ServerAdmin.kt [hunk 2]
```diff
@@ -100,28 +101,65 @@ class ServerAdmin(private val app: App) {
...
+            .url(url)
+            .get()
+        val deployments = JSONArray(executeRequest(request, true))
+        val dep = deployments[0] as JSONObject
+        if (dep.getString("status") != "successful") {
+            RealmLog.error("Failed to deploy: ${dep.toString(4)}")
+        }
+
+        // Work-around for /deployments reporting success, but /register still failing.
+        SystemClock.sleep(5000)
```
Comment (clementetb) [inline realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/admin/ServerAdmin.kt:142]: 🙈

**Review by clementetb — APPROVED** [review]
LGTM

**Comment by cmelchior** [comment]
I'm still debugging this with the Server team, I have a reproduction case, and it very clearly points to a race condition. 5 seconds was bit arbitrary, but I have not seen any errors after adding it. It adds about 30-45 seconds to the total test time. So seems an acceptable work-around until we can get to the bottom of it.

(We need the latest server in order to work on Flexible Sync)

File: realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/admin/ServerAdmin.kt (hunk 1 above)
Comment (cmelchior) [reply realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/admin/ServerAdmin.kt]: Yes, removing it.

**Pull request merged**